### PR TITLE
Use absolute urls for links & relative urls for API requests

### LINKED ISSUE #498: Seed data from the Strapi production instance
https://github.com/iFixit/react-commerce/pull/496 was done to fix the production mini cart checkout, but in the process, we lost relative URLs that made it possible to have a prod preview. This preview was immensely helpful for testing. 

> ~~I'll open another pull to use relative urls in some places, and always us absolute urls in others so that the prod preview works again.~~

We don't want to go back to be using production strapi in a development preview, but we want production data to allow for easier and more confident testing. 

> The problem of seeing the same set of data is solved by improving how we seed the strapi preview instances in Govinor. I'm planning to make them able to seed data from the Strapi production instance, so they'll be virtually the same.

https://github.com/iFixit/react-commerce/issues/498#issuecomment-1195954150

CC: @masonmcelvain @dhmacs

### PR DESCRIPTION
**Pull request by masonmcelvain**
We want to use relative origins on API requests so we don't get blocked
by CORS on previews.

We'll use absolute urls everywhere else, particularly when linking to
other pages. If we try to use a relative url with the JS URL()
constructor, it will throw an error, so we need to use absolute links in
those cases.

## Background

This is meant to have the intended effect of https://github.com/iFixit/react-commerce/pull/393 without causing the bug described here: https://github.com/iFixit/react-commerce/issues/489.

Closes #498

### CODE REVIEW

**Review by andyg0808 — APPROVED** [review]

**Comment by andyg0808** [comment]
CR :dog: through 5c055d2
deploy_block :+1: on a couple comments.

File: frontend/lib/ifixit-api/devices.ts [hunk 1]
```diff
@@ -41,7 +41,9 @@ export function fetchMultipleDeviceImages(
    deviceTitles: string[],
    size: GuideImageSize
 ): Promise<MultipleDeviceApiResponse> {
-   const apiUrl = new URL(`${IFIXIT_ORIGIN}/api/2.0/wikis/topic_images`);
+   const apiUrl = new URL(
+      `${ABSOLUTE_IFIXIT_ORIGIN}/api/2.0/wikis/topic_images`
```
[the remark this answers is not in the document]
Comment (masonmcelvain) [reply frontend/lib/ifixit-api/devices.ts:45]: Good catch! This should be relative

File: frontend/config/env.ts [hunk 2]
```diff
@@ -50,3 +56,21 @@ function checkEnv(env: string | null | undefined, envName: string): string {
...
+   function baseDomain(url: string): string | null {
+      const parsedUrl = new URL(url);
+      const host = parsedUrl.host.match(/[^.]+\.[^.]+$/);
+      return host ? host[0] : '';
+   }
+
+   const currentBaseDomain = baseDomain(window.location.origin);
+   return (
+      Boolean(currentBaseDomain) &&
+      currentBaseDomain === baseDomain(absoluteIfixitOrigin)
```
[the remark this answers is not in the document]
Comment (masonmcelvain) [reply frontend/config/env.ts:74]: For context, this function is copied from https://github.com/iFixit/react-commerce/pull/393.

@danielbeardsley described it as:
> Determine if we should use relatively urls by comapring the base domain
of IFIXIT_ORIGIN and the base domain of the current url. We are using
base domain here so we allow using relative urls when on language
subdomains like es.ifixit.com (where IFIXIT_ORIGIN === [www.ifixit.com](http://www.ifixit.com/)).

Now that we're no longer hosting on a language subdomain, maybe this entire approach doesn't make sense any more though. We want the `vercel.app` previews to be able to talk to ifixit.com/cominor.com, and it seems like this approach won't achieve that.

**Comment by masonmcelvain** [comment]
dev_block ✌🏻

**Comment by sterlinghirsh** [comment]
CR :zap: dev_block :zap: I might suggest keeping it as ifixit origin and putting the relative url logic just in the api client, similar to Danny's pull https://github.com/iFixit/react-commerce/pull/457/files#diff-2cb8839f2fc2e5909e74239210c1db90eecfce6506fecaefbc88d7702e2f261bR48-R55 . This way, all the api logic is kept the same and there's only one place to use this special logic around the origin.

**Comment by masonmcelvain** [comment]
@sterlinghirsh Can do, but does it even make sense to use relative urls anymore if we're no longer hosting on es.ifixit.com? I don't see how using relative urls will help `react-commerce-anything.vercel.app` talk to cominor.com.

**Comment by masonmcelvain** [comment]
I'm gonna close this in light of this comment: https://github.com/iFixit/react-commerce/issues/498#issuecomment-1195954150

**Pull request closed without merging**